int8_t dist0
int8_t dist1
int8_t dist2
int8_t dist3
int8_t dist4
int8_t dist5